routing/route_weight.hpp

```diff
@@ -12,11 +13,26 @@ class RouteWeight final
 public:
   RouteWeight() = default;
 
-  constexpr explicit RouteWeight(double weight) : m_weight(weight) {}
+  constexpr RouteWeight(double weight, int32_t nontransitCross)
```

Comment:
Do you really need int32_t? Why not to use double, for simplicity and uniformity? Or int?

Reply:
Or bool, as it 0 or 1.

Reply:
It's not 0 or 1. It's number of transit-nontransit border crossing, can be 0, 1 or 2 for valid route and can be any inside AStar calculations.

Reply:
Okay, use uint32_t?

Reply:
I'll change to int

Reply:
done